pennylane/debugging.py

```diff
@@ -105,3 +108,75 @@ def get_snapshots(*args, **kwargs):
         return dbg.snapshots
 
     return get_snapshots
+
+
+class PLDB(pdb.Pdb):
+    """Custom debugging class integrated with Pdb."""
+
+    __active_dev = []
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the debugger, and set custom prompt string."""
+        super().__init__(*args, **kwargs)
+        self.prompt = "[pldb]: "
+
+    @classmethod
+    def valid_context(cls):
+        """Determine if the debugger is called in a valid context.
+
+        Raises:
+            TypeError: Can't call breakpoint outside of a qnode execution
+            TypeError: Device not supported with breakpoint
+        """
+
+        if not qml.queuing.QueuingManager.recording() or not cls.is_active_dev():
+            raise TypeError("Can't call breakpoint outside of a qnode execution")
+
+        if cls.get_active_device().name not in ("default.qubit", "lightning.qubit"):
+            raise TypeError("Device not supported with breakpoint")
+
+    @classmethod
+    def add_device(cls, dev):
+        """Add a device to the global active device list.
+
+        Args:
+            dev (Union[Device, "qml.devices.Device"]): The active device
+        """
+        cls.__active_dev.append(dev)
+
+    @classmethod
+    def get_active_device(cls):
+        """Return the active device.
+
+        Raises:
+            ValueError: No active device to get
+
+        Returns:
+            Union[Device, "qml.devices.Device"]: The active device
+        """
+        if not cls.is_active_dev():
+            raise ValueError("No active device to get")
```

Comment:
# Nitpick
Similarly, the issue is not that an argument to this method is incorrect:
```suggestion
            raise RuntimeError("No active device to get")
```

Reply:
Added